flow-defs/carousel.js.flow

```diff
@@ -0,0 +1,71 @@
+// @flow
+
+import * as React from "react";
+declare type PageBulletsProps = {
+  currentIndex: number,
+  numPages: number,
+  onPress?: (index: number) => void,
+};
+declare export var PageBullets: React.ComponentType<PageBulletsProps>;
+declare type ItemsPerPageProp =
+  | {
+      mobile?: number,
+      desktop?: number,
+    }
+  | number;
+declare type MobilePageOffset =
+  | number
+  | {
+      prev: number,
+      next: number,
+    };
+declare type CarouselProps = {
+  items: $ReadOnlyArray<React.Node>,
+  withBullets?: boolean,
+  renderBullets?: (bulletsProps: PageBulletsProps) => React.Node,
+  itemsPerPage?: ItemsPerPageProp,
+  /**
+   * scrolls one page by default
+   */
+  itemsToScroll?: number,
+  /**
+   * number of pixels to show for the next/prev page in mobile
+   */
+  mobilePageOffset?: MobilePageOffset,
+  /**
+   * If true, scroll snap doesn't apply and the user has a free scroll
+   */
+  free?: boolean,
+  autoplay?:
+    | boolean
+    | {
+        time: number,
+      },
+  onPageChange?: (newPageInfo: {
+    pageIndex: number,
+    shownItemIndexes: Array<number>,
+  }) => void,
+};
+declare export var Carousel: React.ComponentType<CarouselProps>;
+declare type CenteredCarouselProps = {
+  items: $ReadOnlyArray<React.Node>,
+  withBullets?: boolean,
```

Comment:
a little detail. Movement of bullets when you click on the right arrow have a little delay. But when you use left arrow the movement is fast as expected

https://user-images.githubusercontent.com/6722153/152204238-ef4729fd-5d91-4721-b48c-495cc3e02b78.mov

Reply:
Yes, good point. That's because we are in page 0 until the carousel scrolls the full page and the 4th item is at the left of the viewport. For this reason the bullet only changes when the full page (3 items) scrolled when going forward, but it changes immediately when we move backward. As an alternative, I could use the middle of the page as a reference, this way the bullets would represent the page with greatest visible percentage and it will change from 0 to 1 when 1.5 items (itemsPerPage/2) are scrolled

Reply:
If the change is not too difficult, we could try this. This is not too important 🍒

Reply:
ok, try it now

Reply:
now is perfect imo!

Reply:
ok, I still need to fix this in the `FullWidthCarousel`

Reply:
should be fixed there too